diag_manager/fms_diag_axis_object.F90

```diff
@@ -337,8 +342,8 @@ end subroutine write_axis_data
   !> @return axis length
   function get_axis_length(this) &
```

Comment:
```
fms_diag_axis_object.F90:352:20:

  352 |       axis_length = this%axis_domain%length(this%cart_name, this%domain_position, this%length)
      |                    1
Error: Reference to impure function 'get_length' at (1) within a PURE procedure

```
this%axis_domain%length is not pure because it is using `mpp_get_compute_domain`